### PR TITLE
Add warning when Settings.setAppElement is not set.

### LINKED ISSUE #780: Check if setAppElement is set on mount
Kind of like `checkProps` that would warn if not set. @donnieberg This would have alerted us to the storybook issue.

Allow Modal to check if global setting is set and require that global is set and tell folks how to set it in the error/warning.

### PR DESCRIPTION
**Pull request by davidlygagnon**
- Adds warning to Modal component to let consumer know when `Settings.setAppElement(el)` was not set prior to using this component.
- Adds prop `ariaHideApp` on `Modal` and `App Laucher` component as a way to opt-out of having to set `Settings.setAppElement(el)`.

Fixes #780 

### Additional description

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [ ] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [ ] `npm run lint:fix` has been run and linting passes.
* [ ] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

**Comment by davidlygagnon** [comment]
This will show the following warning:

screen shot 2018-07-16 at 11 19 59 pm

Just an observation: `React-Modal` also already prints a warning msg when consumers don't use `Modal.setAppElement`. Wouldn't this feel a bit of duplication of warning messages?

I guess the difference here is that we should also let consumers know that `Settings.setAppElement(el)` wasn't set and should have been.

File: components/app-launcher/index.jsx [hunk 1]
```diff
@@ -31,6 +31,7 @@ const defaultProps = {
 	assistiveText: {
 		trigger: 'Open App Launcher',
 	},
+	ariaHideApp: true,
```
Comment (davidlygagnon) [inline components/app-launcher/index.jsx]: React modal has this prop to "opt out" of setAppElement. I think we should use that as well.

File: components/modal/index.jsx [hunk 2]
```diff
@@ -181,6 +188,9 @@ class Modal extends React.Component {
 	componentWillMount () {
 		this.generatedId = shortid.generate();
 		checkProps(MODAL, this.props);
+		if (this.props.ariaHideApp) {
+			checkAppElementIsSet();
```
Comment (davidlygagnon) [inline components/modal/index.jsx]: If consumer hasn't opted-out, check if the app element was set. The app-launcher component also uses this modal internally so I only need to add this check here.

**Comment by interactivellama** [comment]
I'd like to not have truthy default props, but since this used in React Modal and I hope "no one disables it ever." This looks good.

**Pull request merged**